Sources/MUXSDKStatsObjc/MUXSDKMonitor.h

```diff
@@ -0,0 +1,397 @@
+/// does nothing. Automatic video change detection is enabled
+/// by default.
+///
+/// - Parameter playerName: The name of the player for which
+/// automatic video change detection should be enabled.
+- (void)enableAutomaticVideoChangeDetectionForPlayerName:(nonnull NSString *)playerName;
+
+/// Disables automatic video change detection, if automatic
+/// video change detection is already disabled then this method
+/// does nothing. Automatic video change detection is enabled
+/// by default.
+///
+/// - Parameter playerName: The name of the player for which
+/// automatic video change detection should be disabled.
+- (void)disableAutomaticVideoChangeDetectionForPlayerName:(nonnull NSString *)playerName;
+
+#pragma mark - Manual Video Change
+
+
+/// Signals that a player is now playing a different video.
+///
+/// Use this method to signal that the player is now playing
+/// a new video. The player name must previously have been
+/// passed in a `startMonitoring*` call. If the player name 
+/// hasn't been previously passed, an exception will be raised.
+///
+/// - Parameters:
+///   - playerName: The name of the player now playing a
+///   different video.
+///   - customerData: A MUXSDKCustomerData object with player,
+///   video, and view metadata
+- (void)signalVideoChangeForPlayerName:(nonnull NSString *)playerName
+               withUpdatedCustomerData:(nullable MUXSDKCustomerData *)customerData;
+
+#pragma mark - Program Change
+
+/// Signals that a player is now playing a different video
+/// of a playlist; or a different program of a live stream
+///
+/// Use this method to signal that the player is now playing
+/// a different video of a playlist, or a different program
+/// of a live stream. The player name must previously have been
+/// passed in a `startMonitoring*` call. If the player name
+/// hasn't been previously passed, an exception will be raised.
+///
+/// - Parameters:
+///   - playerName: The name of the player now playing a 
+///   different video of a playlist; or a different program
+///   of a live stream
+///   - customerData: A MUXSDKCustomerData object with player,
+///   video, and view metadata
+- (void)signalProgramChangeForPlayerName:(nonnull NSString *)playerName
+                 withUpdatedCustomerData:(nullable MUXSDKCustomerData *)customerData;
+
+#pragma mark - Customer Data
+
+/// Allows customerData to be set or updated for a player
+/// instance
+///
+/// - Parameters:
+///   - playerName: The name of the player for which to update
+///   the customer data
+///   - customerData: A MUXSDKCustomerData object with player,
+///   video, and view metadata
+- (void)updatePlayerName:(nonnull NSString *)playerName
+        withCustomerData:(nullable MUXSDKCustomerData *)customerData;
+
+#pragma mark - Orientation Change
+
+// TODO: Move MUXSDKViewOrientation to its own header
+
+// TODO: We should probably allow a starting orientation to be set when calling startMonitoring on an AVPlayer
```

Comment:
@jsanford8 what do you think? When `startMonitoring` is called for an `AVPlayer` I don't think there is a way for us to detect its orientation. (I need to confirm this.)

Reply:
Yes, that would probably be a good thing. The other option (what people do today) is fire an orientationchange event right away to signal the first orientation, which should work.

Reply:
Yeah I think the other option sounds like the way to go for now, to avoid multiplying the different permutations of parameters on these `startMonitoring` methods. Sending an orientation change right after calling `startMonitoring` should still work the same way it did before.

At some point I think it would make sense to either add something similar to `MUXSDKCustomerData` but for parameters specific to this layer or extend `MUXSDKCustomerData` to include these additional parameters.